Начнем с двуногой лестницы при U = 0. Гамильтониан без взаимодействия будет иметь вид:

$$\hat{H}= -t\sum_{i} {a_{i,1}^+}{a_{i+1,1}} + h.c.  -t\sum_{i} {a_{i,2}^+}{a_{i+1,2}} + h.c. -t_{+}\sum_{i} {a_{i,1}^+}{a_{i,2}} + h.c.$$
    
Здесь h.c. - эрмитово сопряжение

Перейдем к импульсам с помощью преобразования Фурье:
$$a_k^+= \frac{1}{\sqrt{L}}\sum_{x} e^{ikx}a_x^+$$

$$a_k= \frac{1}{\sqrt{L}}\sum_{x} e^{-ikx}a_x$$

$$a_x^+= \frac{1}{\sqrt{L}}\sum_{k} e^{-ikx}a_k^+$$

$$a_x= \frac{1}{\sqrt{L}}\sum_{k} e^{ikx}a_k$$

Энергия одной частицы вдоль одной ноги 

$$\epsilon_k = -2tcos(ka)$$

Но поскольку энергия определена с точностью до константы, будем использовать:

$$\epsilon_k = 2t(1-cos(ka))$$

Тогда наш гамильтониан примет вид:

$$\hat{H}= \sum_{k_1}2t(1-cos(k_1a)){a_{k,1}^+}{a_{k,1}} + \sum_{k_2}2t(1-cos(k_2a)){a_{k,2}^+}{a_{k,2}} - t_{+}(\sum_{k} {a_{k,1}}^+{a_{k,2}} + h.c.)$$

Это выражение можно диагонализовать следующим образом:

$$\begin{pmatrix}{a_{k,1}^+}& {a_{k,2}^+}\end{pmatrix}\begin{pmatrix} \epsilon_k& -t_{+}\\-t_{+}& \epsilon_k\end{pmatrix}\begin{pmatrix}{a_{k,1}}\\{a_{k,2}}\end{pmatrix} = \begin{pmatrix}{c_{k,1}^+}& {c_{k,2}^+}\end{pmatrix}\begin{pmatrix} \lambda_1& 0\\0& \lambda_2\end{pmatrix}\begin{pmatrix}{c_{k,1}}\\{c_{k,2}}\end{pmatrix}$$

Остается найти собственные значения матрицы коэффициентов слева. 

$$\begin{vmatrix} \epsilon_k-\lambda& -t_{+}\\-t_{+}& \epsilon_k-\lambda\end{vmatrix}=0$$

$$(\epsilon_k-\lambda)^2 - (t_{+})^2 = 0$$

$$\lambda_{1,2} = \epsilon_k \pm t_{+}$$

Тогда итоговый гамильтониан имеет вид (в нем как раз видна энергия одной частицы):
    
$$\hat{H}= \sum_{k}(\epsilon_k-t_{+}){c_{k,1}^+}{c_{k,1}} + \sum_{k}(\epsilon_k+t_{+}){c_{k,2}^+}{c_{k,2}}= \\ = \sum_{k}(2t(1-cos(ka))-t_{+}){c_{k,1}^+}{c_{k,1}} + \sum_{k}(2t(1-cos(ka))+t_{+}){c_{k,2}^+}{c_{k,2}}$$

Как следует из [статьи](https://arxiv.org/pdf/cond-mat/0605350.pdf) Е.А. Буровского с коллегами формула (2.3), энергия двух частиц на решетке будет равна:

$${U^*}^{-1} = -\int_{1BZ}\frac{dk}{(2\pi)^3}\frac{1}{2\epsilon_k}$$

при выборе энергии:

$$\epsilon_k = 2t(1-cos(ka))$$

и  $$a = 1$$ 

Интеграл по первой зоне Бриллюэна по каждой из переменных нужно брать на отрезке:

$$[-\frac{\pi}a;\frac{\pi}a]$$

будет равна

$${U^*} = -7.915t$$

В качестве теста можно попробовать взять этот интеграл численно для *t = a = 1*, там, конечно, есть особые точки (когда знаменатель равен нулю), но поскольку наш приближенный метод в эти точки почти наверное не попадет, мы этот интеграл как бы берем в смысле главного значения:

In [12]:
from math import pi, cos
import numpy as np

def foo(x,y,z):
    return 1/(2*((2*pi)**3) * (
    2*(1-cos(x))+2*(1-cos(y))+2*(1-cos(z))
    ))

In [15]:
X = np.linspace(-pi, pi, 3000)
Y = np.linspace(-pi, pi, 3000)
Z = np.linspace(-pi, pi, 3000)
h = 2*pi/3000
integral = 0

In [ ]:
for i in enumerate(X):
    for j in enumerate(Y):
        for k in enumerate(Z):
            integral+=(h**3)*foo(i,j,k)

In [ ]:
Правило такого подсчета интеграла, например, [отсюда]()

Я считал это в другом ноутбуке, считается очень долго, но ответ такой:

    integral = 0.12630106997797813

In [28]:
U = 1/0.12630106997797813
print(U)

7.917589298129937


Если считать кубиком $3000*3000*3000$ при $a = 1$, $U = - 7.917589298129937 * t$

Посчитаем наш интеграл аналогичным образом:

$${U^*}^{-1} = -\int_{1BZ}\frac{dk_1 dk_2}{(2\pi)^2}\frac{1}{2((2t(1-cos(k_1a))-t_{+}) + (2t(1-cos(k_2a))+t_{+}))}$$

In [23]:
def _2_leg_energy(k1, k2, t, t_plus):
    return 1/(2*((2*pi)**2) * (
    (2*t*(1-cos(k1))-t_plus)+(2*t*(1-cos(k2))+t_plus)
    ))

Будем, на всякий случай, считать, что $t = t_+ = 1$

Переопределим функцию, чтоб попроще считалось:

In [44]:
def _2_leg_energy_simplified(k1, k2):
    return 1/(2*((2*pi)**2) * (
    2*(1-cos(k1))-1 + 2*(1-cos(k2))+1
    ))

In [45]:
_2_leg_energy_simplified(1,2)

0.003375852248988832

In [49]:
from datetime import datetime as dt

X = np.linspace(-pi, pi, 3000)
Y = np.linspace(-pi, pi, 3000)

h = 2*pi/3000
integral = 0

In [50]:
print(dt.now().ctime())
for i in X:
    for j in Y:
        integral+=(h**3)*_2_leg_energy_simplified(i,j)
print(dt.now().ctime())

Thu Oct 15 23:29:17 2020
Thu Oct 15 23:29:23 2020


In [52]:
integral

0.0015000537368560912

Для трехногой лестницы гамильтониан без взаимодействия будет иметь следующий вид:

$$\hat{H}= -t\sum_{i} {a_{i,1}^+}{a_{i+1,1}} + h.c.  -t\sum_{i} {a_{i,2}^+}{a_{i+1,2}} + h.c. -t\sum_{i} {a_{i,3}^+}{a_{i+1,3}} + h.c. -t_{+}\sum_{i} {a_{i,1}^+}{a_{i,2}} + h.c. -t_{+}\sum_{i} {a_{i,2}^+}{a_{i,3}} + h.c.$$

Используя то же Фурье-преобразование, переходим к *k*. Свободная энергия "вдоль" ноги лестницы остается такой же. Гамильтониан в испульсном представлении принимает вид:

$$\hat{H}= \sum_{k}2t(1-cos(ka)){a_{k,1}^+}{a_{k,1}} + \sum_{k}2t(1-cos(ka)){a_{k,2}^+}{a_{k,2}} +\sum_{k}2t(1-cos(ka)){a_{k,3}^+}{a_{k,3}} - t_{+}(\sum_{k} {a_{k,1}}^+{a_{k,2}} + h.c. + {a_{k,2}}^+{a_{k,3}} + h.c.)$$



Диагонализацию проводим так же:

$$ \begin{pmatrix}{a_{k,1}^+}& {a_{k,2}^+}& {a_{k,3}^+}\end{pmatrix}\begin{pmatrix}\epsilon_k & -t_{+} & 0 \\-t_{+} & \epsilon_k & -t_{+} \\ 0 & -t_{+} & \epsilon_k \end{pmatrix}\begin{pmatrix}{a_{k,1}} \\ {a_{k,2}} \\ {a_{k,3}} \end{pmatrix} = \begin{pmatrix}{c_{k,1}^+}& {c_{k,2}^+}& {c_{k,3}^+}\end{pmatrix}\begin{pmatrix} \lambda_1& 0 & 0\\0& \lambda_2 & 0 \\ 0 & 0 & \lambda_3\end{pmatrix}\begin{pmatrix}{c_{k,1}} \\ {c_{k,2}} \\ {c_{k,3}} \end{pmatrix}$$

$$\begin{vmatrix}\epsilon_k - \lambda & -t_{+} & 0 \\-t_{+} & \epsilon_k - \lambda& -t_{+} \\ 0 & -t_{+} & \epsilon_k - \lambda \end{vmatrix} = 0$$

Разложим по первой строке:

$$ (\epsilon_k - \lambda)((\epsilon_k - \lambda)^2 - {t_{+}}^2) + {t_{+}}(-{t_{+}}(\epsilon_k - \lambda)) = 0$$

Вынесем общий множитель:

$$ (\epsilon_k - \lambda)((\epsilon_k - \lambda)^2 - 2{t_{+}}^2)= 0$$

Соответственно:
$$\lambda_1 = \epsilon_k\\
\lambda_2 = \epsilon_k + {t_{+}}\sqrt{2}\\
\lambda_3 = \epsilon_k - {t_{+}}\sqrt{2}$$

Тогда итоговый гамильтониан имеет вид (в нем как раз видна энергия одной частицы):
    
$$\hat{H}= \sum_{k}\epsilon_k{c_{k,1}^+}{c_{k,1}} + \sum_{k}(\epsilon_k + {t_{+}}\sqrt{2}){c_{k,2}^+}{c_{k,2}} + \sum_{k}(\epsilon_k - {t_{+}}\sqrt{2}){c_{k,3}^+}{c_{k,3}} = \\ = \sum_{k}(2t(1-cos(ka))){c_{k,1}^+}{c_{k,1}} + + \sum_{k}(2t(1-cos(ka)) + {t_{+}}\sqrt{2}){c_{k,2}^+}{c_{k,2}} + \sum_{k}(2t(1-cos(ka)) - {t_{+}}\sqrt{2}){c_{k,3}^+}{c_{k,3}}$$

Посчитаем наш интеграл аналогичным образом:

$${U^*}^{-1} = -\int_{1BZ}\frac{dk_1 dk_2 dk_3}{(2\pi)^3}\frac{1}{2(2t(1-cos(k_1a)) + (2t(1-cos(k_2a)) - {t_{+}}\sqrt{2}) + (2t(1-cos(k_3a)) - {t_{+}}\sqrt{2}))}$$